index.bs

```diff
@@ -261,11 +269,37 @@ The <dfn method for=FileSystemFileHandle>getFile()</dfn> method, when invoked, m
 
 <div class="note domintro">
   : |writer| = await |fileHandle| . {{FileSystemFileHandle/createWriter()}}
-  :: Returns a {{FileSystemWriter}} that can be used to write to the file.
+  : |writer| = await |fileHandle| . {{FileSystemFileHandle/createWriter()|createWriter}}({ {{FileSystemCreateWriterOptions/atomic}}: true })
+  : |writer| = await |fileHandle| . {{FileSystemFileHandle/createWriter()|createWriter}}({ {{FileSystemCreateWriterOptions/truncate}}: |truncate|, {{FileSystemCreateWriterOptions/atomic}}: true })
+  :: Returns a {{FileSystemWriter}} that can be used to write to the file. Any changes made through
+     |writer| won't be reflected in the file represented by |fileHandle| until its
+     {{FileSystemWriter/close()}} or {{FileSystemWriter/flush()}} methods are called.
+     User agents try to ensure that no partial writes happen, i.e. the file represented by
+     |fileHandle| will either contains its old contents or it will contain whatever data was written
+     through |writer| up until one of these methods was called.
+
+     This is typically implemented by writing data to a temporary file, and only replacing the file
+     represented by |fileHandle| with the temporary file when the writer is closed.
+
+     If |truncate| is `true` or not specified, the temporary file starts out empty, otherwise the

```

Comment:
Instead of a flag, how about an enum?

`fopen` provides the following options closest to what we want to achieve:
* `w` for truncate mode
* 'r+` for read/write, with the cursor at the beginning of the file

I like the idea of `update` to be the opposite of `truncate`.

If we want to contrast with `fopen` so there's no confusion, we can use slightly different enums:
```webidl

enum FileSystemWriterMode {"truncate", "update"};

dictionary FileSystemCreateWriterOptions {
  FileSystemWriterMode mode = "truncate";
  boolean atomic = true;
};
```

Reply:
I went with victor's suggested names for now.